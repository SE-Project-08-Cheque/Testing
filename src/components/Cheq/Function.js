function toWords(s=""){
    if(parseInt(s)>10**15){
        return 'Value is too long to display'
    }
    var th = ['','thousand','million', 'billion','trillion'];
    var dg = ['zero','one','two','three','four', 'five','six','seven','eight','nine'];
    var tn = ['ten','eleven','twelve','thirteen', 'fourteen','fifteen','sixteen', 'seventeen','eighteen','nineteen'];
    var tw = ['twenty','thirty','forty','fifty', 'sixty','seventy','eighty','ninety'];
    s = s.toString();
    s = s.replace(/[\, ]/g,'');
    if (s != parseFloat(s)) return '';
    s=parseFloat(s).toFixed(2);
    var x = s.indexOf('.');
    if (x == -1)
        x = s.length;
    if (x > 15){
        return 'Value is too long to display';
    }
    var n = s.split(''); 
    var str = '';
    var sk = 0;
    for (var i=0;   i < x;  i++) {
        if ((x-i)%3==2) { 
            if (n[i] == '1') {
                str += tn[Number(n[i+1])] + ' ';
                i++;
                sk=1;
            } else if (n[i]!=0) {
                str += tw[n[i]-2] + ' ';
                sk=1;
            }
        } else if (n[i]!=0) { // 0235
            str += dg[n[i]] +' ';
            if ((x-i)%3==0) str += 'hundred ';
            sk=1;
        }
        if ((x-i)%3==1) {
            if (sk)
                str += th[(x-i-1)/3] + ' ';
            sk=0;
        }
    }
    str= str.trim()+' Rupees '
    if (x != s.length) {
    n=n.slice(x+1,n.length);
    var m = [...new Set(n)];
    if((m.length==1 && m.includes("0")) || m.length==0){
      return str+'Only';
    }
    x=n.length;
    for (var i=0;   i < x;  i++) {
      if ((x-i)%3==2 || x==1) { 
          if (n[i] == '1') {
              str += tn[Number(n[i+1])] + ' ';
              i++;
              sk=1;
          } else if (n[i]!=0) {
              str += tw[n[i]-2] + ' ';
              sk=1;
          }
      } else if (n[i]!=0) {
          str += dg[n[i]] +' ';
          sk=1;
      }
  }
  str=str+'Cents Only'
  }
  else{
    str=str+'Only'
    }
      return str;
    }

function format(num){
    num=parseFloat(num);
    if (num !== parseFloat(num)) return '0.00';
    var n =num.toFixed(2);
    return String(n).replace(/(?<!\..*)(\d)(?=(?:\d{3})+(?:\.|$))/g, '$1,')
    }
export {toWords,format}