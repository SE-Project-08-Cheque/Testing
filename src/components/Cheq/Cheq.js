// import { border } from '@chakra-ui/react';
import React from 'react';
import cheq from './Cheq.module.css';
import {toWords,format} from './Function'

const cheque = (props) => (
  <div data-testid='cheque' className={cheq.check}>
    <div className={cheq.border}>
      <div className={cheq.container}>
      {/* <Cross crosstxt crosstxt={'ONLY SRI LANKA'}/> */}
        <div className={cheq.content}>
          <div className={cheq.one}>
            <div className={cheq.title}>
              <div className={cheq.ownerName}>{props.name}</div>
              <div className={cheq.name}>
                {props.designation}
                <br />{props.company}
                <br />{props.city}
              </div>
            </div>
            <div className={cheq.following}>
              <table className={cheq.table}>
                <tbody>
                  <tr>
                    <td className={cheq.line} >
                      This check is in payment of the following
                    </td>
                  </tr>
                  <tr>
                    <td className={[cheq.empty,cheq.line].join(" ")}>{props.reason1}</td>
                  </tr>
                  <tr>
                    <td className={[cheq.empty,cheq.line].join(" ")}>{props.reason2}</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div className={cheq.date}>{props.date}</div>
          </div>
          <div className={cheq.one}>
            <div className={cheq.title}>
            <div className={cheq.payto}>{props.payto}</div>
          <div className={cheq.orderof}>{toWords(props.amount)} rupees only</div>
 
            </div>
            <div className={cheq.following}>
              
            </div>
            <div className={cheq.adt}>
                <fieldset style={{marginTop:'-15px'}}>
                <legend><div className={cheq.amounttxt}>amount</div></legend>
                    <div style={{fontSize:'12px'}}>LKR</div>
                <div className={cheq.t}>
                    {format(props.amount)}
                </div>
                </fieldset>
                
            </div>
          </div>

          <div className={cheq.three}>
            <div className={cheq.company}>
              <div className={cheq.companyName}>OUR BANK</div>
              <div className={cheq.companyDetails}>
                OUR BANK <br/>
                COMPANY ADDRESS
                <br /> CITY, XIP
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
);
export default cheque;
    