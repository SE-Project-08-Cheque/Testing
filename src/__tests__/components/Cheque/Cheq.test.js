import React from 'react'
import ReactDOM from 'react-dom'
import Cheq from '../../../components/Cheq/Cheq'
import { BrowserRouter } from "react-router-dom";
import {render,cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import renderer from 'react-test-renderer'
// import {} from 'jest-dom'
afterEach(cleanup)
it("Chequeform renders without crashing",()=>{
    const div=document.createElement("div")
    ReactDOM.render(<Cheq/>,div)
})
it("Chequeform renders without crashing",()=>{
    const {getByTestId} = render(<Cheq 
        name={"Dakshitha"}
        designation={"Manager"}
        company={"CIC"}
        city={"Colombo"}
        reason1={"To Buy"}
        reason2={""}
        amount={"5000"}
        payto={"8056738"} />)
    expect(getByTestId('cheque')).toHaveTextContent("Dakshitha")
    expect(getByTestId('cheque')).toHaveTextContent("Manager")
    expect(getByTestId('cheque')).toHaveTextContent("CIC")
    expect(getByTestId('cheque')).toHaveTextContent("Colombo")
    expect(getByTestId('cheque')).toHaveTextContent("To Buy")
    expect(getByTestId('cheque')).toHaveTextContent("5,000.00")
})
it("matches snapshot",()=>{
    const tree=renderer.create(<Cheq/>).toJSON()
    expect(tree).toMatchSnapshot();
})