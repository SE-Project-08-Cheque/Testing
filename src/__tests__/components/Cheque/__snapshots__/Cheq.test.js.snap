// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="check"
  data-testid="cheque"
>
  <div
    className="border"
  >
    <div
      className="container"
    >
      <div
        className="content"
      >
        <div
          className="one"
        >
          <div
            className="title"
          >
            <div
              className="ownerName"
            />
            <div
              className="name"
            >
              <br />
              <br />
            </div>
          </div>
          <div
            className="following"
          >
            <table
              className="table"
            >
              <tbody>
                <tr>
                  <td
                    className="line"
                  >
                    This check is in payment of the following
                  </td>
                </tr>
                <tr>
                  <td
                    className="empty line"
                  />
                </tr>
                <tr>
                  <td
                    className="empty line"
                  />
                </tr>
              </tbody>
            </table>
          </div>
          <div
            className="date"
          />
        </div>
        <div
          className="one"
        >
          <div
            className="title"
          >
            <div
              className="payto"
            />
            <div
              className="orderof"
            >
              
               rupees only
            </div>
          </div>
          <div
            className="following"
          />
          <div
            className="adt"
          >
            <fieldset
              style={
                Object {
                  "marginTop": "-15px",
                }
              }
            >
              <legend>
                <div
                  className="amounttxt"
                >
                  amount
                </div>
              </legend>
              <div
                style={
                  Object {
                    "fontSize": "12px",
                  }
                }
              >
                LKR
              </div>
              <div
                className="t"
              >
                0.00
              </div>
            </fieldset>
          </div>
        </div>
        <div
          className="three"
        >
          <div
            className="company"
          >
            <div
              className="companyName"
            >
              OUR BANK
            </div>
            <div
              className="companyDetails"
            >
              OUR BANK 
              <br />
              COMPANY ADDRESS
              <br />
               CITY, XIP
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
